### PR TITLE
Fix* emissions reporting

### PR DESCRIPTION
**Pull request by eringold**
Pull request overview
---------------------

\*Not quite a fix to https://github.com/NREL/ComStock/issues/220 pending EnergyPlus version update. Until we upgrade to >24.2, the 'out.emissions.electricity.enduse_group.hvac.*' columns will be zero for almost all models. However, the 'calc.weighted.enduse_group.electricity.hvac.emissions.egrid_2021_subregion' column is still calculated in postprocessing from HVAC energy and total electricty emissions instead of the enduse_group as reported in the measure. 

Once we use EnergyPlus with the fixed custom meters, we can use the reported HVAC emissions directly to calculate weighted enduse_group emissions.

This PR:
- better aligns the emissions reporting with energy - instead of a 'CoolingHVAC' custom meter with cooling and HeatRejection, reports cooling and heating end-uses, and total HVAC including heating, cooling and associated energy
- improves the parsing of registered value names to column definitions, including field_description column
- updates measure test model with newer/smaller/cleaner one
- a couple small changes to utility bills measure to have it always register a value for fuel dollars. For small runs that might not have fuel use/energy costs, no registered values means no columns are created. But comstockpostproc uses the full list of possible energy cost column names (e.g. [here](https://github.com/NREL/ComStock/blob/92f46d3e4647483b48bbfe3ab76ca75199bd4a7d/postprocessing/comstockpostproc/comstock.py#L2769)), so having the measure always register a value even if zero will avoid having to create them in postprocessing.
- when generating a sample to test this change locally, ran into small issue running Parallel in tsv_resampling.py on my machine. It's happened before, and `prefer='threads'` fixes it. 
- added some notes on setting up env/running sampling in the sampling readme. @rHorsey todo.

### Pull Request Author

This pull request makes changes to (select all the apply):
 - [ ] Documentation
 - [ ] Infrastructure (includes apptainer image, buildstock batch, dependencies, continuous integration tests)
 - [x] Sampling (sort of/not really)
 - [ ] Workflow Measures
 - [ ] Upgrade Measures
 - [x] Reporting Measures
 - [x] Postprocessing

Author pull request checklist:

 - [x] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [ ] Added tests for new measures
 - [x] Updated measure .xml(s)
 - [x] Register values added to `comstock_column_definitions.csv`
 - [ ] Both `options_lookup.tsv` files updated
 - [ ] 10k+ test run
 - [ ] Change documentation written
 - [ ] Measure documentation written
 - [ ] ComStock documentation updated
 - [ ] Changes reflected in example `.yml` files
 - [ ] Changes reflected in `README.md` files
 - [ ] Added 'See ComStock License' language to first two lines of each code file
 - [ ] Implements corresponding measure tests and indexing path in `test/measure_tests.txt` or/and `test/resource_measure_tests.txt`
 - [ ] All new and existing tests pass the CI

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] Reviewed change documentation
 - [ ] Ensured code files contain License reference
 - [ ] Results differences are reasonable
 - [ ] Make sure the newly added measures has been added with tests and indexed properly
 - [ ] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```

### CODE REVIEW

**Review by asparke2 — DISMISSED** [review]
I fixed merge conflicts with the column definitions.csv.

**Review by asparke2 — DISMISSED** [review]
I reviewed and fixed merge conflicts in the column definition csv.

**Review by asparke2 — APPROVED** [review]

File: postprocessing/comstockpostproc/resources/comstock_column_definitions.csv [hunk 1]
```diff
@@ -647,297 +647,285 @@ results.csv,com_stock_sensitivity_reports.com_report_smallest_space_m_2,out.para
...
-results.csv,emissions_reporting.annual_electricity_ghg_emissions_egrid_2021_state_kg,out.emissions.electricity.egrid_2021_state,TRUE,FALSE,float,co2e_kg,co2e_kg,"annual greenhouse gas emissions from electricity use, using egrid 2021 state emissions intensity"
-results.csv,emissions_reporting.annual_electricity_ghg_emissions_egrid_2021_subregion_kg,out.emissions.electricity.egrid_2021_subregion,TRUE,FALSE,float,co2e_kg,co2e_kg,"annual greenhouse gas emissions from electricity use, using egrid 2021 subregion emissions intensity"
-results.csv,emissions_reporting.winter_daily_average_electricity_ghg_emissions_egrid_2021_state_kg,out.emissions.electricity.winter_daily_average.egrid_2021_state,TRUE,FALSE,float,co2e_kg,co2e_kg,"winter daily average greenhouse gas emissions from electricity use, using egrid 2021 state intensity"
-results.csv,emissions_reporting.winter_daily_average_electricity_ghg_emissions_egrid_2021_subregion_kg,out.emissions.electricity.winter_daily_average.egrid_2021_subregion,TRUE,FALSE,float,co2e_kg,co2e_kg,"winter daily average greenhouse gas emissions from electricity use, using egrid 2021 subregion intensity"
```
Comment (JieXiong9119) [inline postprocessing/comstockpostproc/resources/comstock_column_definitions.csv:664]: @eringold I noticed you deleted these "daily_average" emissions entries. They were added specifically for the emission objective lighting measure recently. Could we add them back?

Comment (asparke2) [reply postprocessing/comstockpostproc/resources/comstock_column_definitions.csv:664]: @JieXiong9119 this was my mistake, I missed those in the conflict resolution in this file. I can make a PR can fix tomorrow or Thurs.

Comment (JieXiong9119) [reply postprocessing/comstockpostproc/resources/comstock_column_definitions.csv:664]: Thanks! FYI, I believe there are 15 rows with substring `daily_average_electricity_ghg_emission`.

**Pull request merged**